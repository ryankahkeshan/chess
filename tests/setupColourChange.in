setup
+ K e1
+ k e8
+ p a1
+ p g8
+ r h8
= black
done
- a1
done
- g8
done
game human human
move e1 e2
move e8 g8
resign
