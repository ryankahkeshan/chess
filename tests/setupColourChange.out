8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 p _ K _ 

  abcdefgh

8  _ _k_p_
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 p _ K _ 

  abcdefgh

8  _ _k_pr
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 p _ K _ 

  abcdefgh

Invalid Setup: 
  - Each player must have exactly one king
  - There should be no pawns on the first or last row
8  _ _k_pr
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid Setup: 
  - Each player must have exactly one king
  - There should be no pawns on the first or last row
8  _ _k_ r
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

8  _ _k_ r
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Invalid move
8  _ _ rk_
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh

Black wins!
Final Score:
White: 0
Black: 1
